src/Spectre.libHeatmapDataScaling/HistogramEqualization.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+* HistogramEqualization.cpp
+* Class with histogram equalization algorithm implementation for contrast enhancement.
+*
+Copyright 2018 Daniel Babiak
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "Spectre.libHeatmapDataScaling/HistogramEqualization.h"
+
+namespace spectre::visualization
+{
+std::vector<double> HistogramEqualization::scaleData(const gsl::span<double> intensities)
+{
+    std::vector<double> roundedIntensities = roundIntensities(intensities);
+    std::map<double, unsigned int> const histogramMap = calculateCumulativeDistribution(countRepeatingValues(roundedIntensities));
+    return calculateNewHistogramData(roundedIntensities, histogramMap);
+}
+
+std::vector<double> HistogramEqualization::roundIntensities(const gsl::span<double> intensities)
+{
+    std::vector<double> roundedIntensities(intensities.size());
+    for (int i = 0; i < intensities.size(); i++)
+    {
+        roundedIntensities[i] = round(intensities[i]);
```

Comment:
You're using these rounded values to map them to counts in some ranges. I'm not sure however, what will be in `roundedIntensities` after simple rounding - we still may have some slight differences, due to `double` precision, depending on scale of input data. Can't we go more stable and predictable way of realizing the same through `roundedIntensities[i] = static_cast<int>(intensities[i] + 0.5);`? @Sand3r-, I think you could validate my idea.

I am just not sure, how using `map` with `double` keys will end.